src/main/java/com/minecolonies/client/gui/WindowHutFarmer.java

```diff
@@ -1,243 +1,316 @@
+        fields = building.getFields();
     }
 
-
     /**
-     * Action performed when carrot button is clicked
+     * Fired when the assignment mode has been toggled.
      *
-     * @param ignored   Parameter is ignored, since some actions require a button.
-     *                  This method does not
+     * @param button clicked button.
      */
-    private void carrotClicked(Button ignored)
+    private void assignmentModeClicked(@NotNull Button button)
     {
-        if (building.carrot >= MAX_AMOUNT)
+        if (button.getLabel().equals(LanguageHandler.format("com.minecolonies.gui.hiring.off")))
         {
-            return;
+            button.setLabel(LanguageHandler.format("com.minecolonies.gui.hiring.on"));
+            building.setAssignFieldManually(true);
         }
-        building.carrot++;
-        removeOthers(CARROT);
+        else
+        {
+            button.setLabel(LanguageHandler.format("com.minecolonies.gui.hiring.off"));
+            building.setAssignFieldManually(false);
+        }
+        window.findPaneOfTypeByID(LIST_FIELDS, ScrollingList.class).refreshElementPanes();
     }
 
+    @NotNull
     @Override
     public String getBuildingName()
     {
-        return "com.minecolonies.gui.workerHuts.farmer";
+        return "tile.minecolonies.blockHutFarmer.name";
     }
 
     @Override
     public void onOpened()
     {
         super.onOpened();
 
-        updateButtonLabels();
+        if (building.assignFieldManually())
+        {
+            findPaneOfTypeByID(TAG_BUTTON_ASSIGNMENT_MODE, Button.class).setLabel(LanguageHandler.format("com.minecolonies.gui.hiring.on"));
+        }
+        else
+        {
+            findPaneOfTypeByID(TAG_BUTTON_ASSIGNMENT_MODE, Button.class).setLabel(LanguageHandler.format("com.minecolonies.gui.hiring.off"));
+        }
+
         findPaneOfTypeByID(BUTTON_PREV_PAGE, Button.class).setEnabled(false);
         buttonPrevPage = findPaneOfTypeByID(BUTTON_PREV_PAGE, Button.class);
         buttonNextPage = findPaneOfTypeByID(BUTTON_NEXT_PAGE, Button.class);
+
+        fieldList = findPaneOfTypeByID(LIST_FIELDS, ScrollingList.class);
+        fieldList.setDataProvider(new ScrollingList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return fields.size();
+            }
+
+            @Override
+            public void updateElement(int index, @NotNull Pane rowPane)
+            {
+                final FieldView field = fields.get(index);
+                @NotNull final String distance = Integer.toString((int) Math.sqrt(BlockPosUtil.getDistanceSquared(field.getId(), building.getLocation())));
+                final String direction = calcDirection(building.getLocation(), field.getId());
+                @NotNull final String owner = field.getOwner().isEmpty() ? ("<" + LanguageHandler.format("com.minecolonies.gui.workerHuts.farmerHut.unused") + ">") : field.getOwner();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 183 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00103)